### DISCUSSION TITLE
EMQX ACL Deny behaviour - MQTTv5

### DISCUSSION THREAD

**Discussion by ennc0d3** [emqx/emqx#13239] [Q&A] [answered]
I am using EMQX v5.7.0 and trying to subscribe to a topic (MQTTv5 client) which is denied in Authorization (File setting), the subscription attempted is denied as expected but EMQX is not sending SUBACK with reason code, instead it is sending DISCONNECT

Is there a way to configure it to send SUBACK instead of DISCONNECT? So, as I client i know why it is actually disconnected.

**Comment by ieQu1** [comment]
https://docs.emqx.com/en/emqx/latest/access-control/authz/authz.html#configure-with-configuration-file

What is the value of `deny_action`?

**Reply by ennc0d3** [reply]
Its set to ```deny_action = disconnect``` but the possible values are ignore and disconnect, so I am thinking how could i configure the rule to send suback with rejected reason?

**Reply by ieQu1** [reply, marked as answer]
That's the behavior of `deny_action` = `ignore` for SUBSCRIBE ~with QoS > 0~

**Reply by zmstone** [reply]
> That's the behavior of `deny_action` = `ignore` for SUBSCRIBE with QoS > 0

~This is new to me, checked the code to confirm, I cannot seem to find where in the spec says the server should or should not send SUBACK depending on QoS.~

Ignore me. I was looking at PUBACK code.

**Reply by zmstone** [reply]
> When the Server receives a SUBSCRIBE packet from a Client, the Server MUST respond with a SUBACK packet [MQTT-3.8.4-1]. The SUBACK packet MUST have the same Packet Identifier as the SUBSCRIBE packet that it is acknowledging [MQTT-3.8.4-2].

**Reply by ieQu1** [reply]
You are right, thanks for the correction.
